repose-aggregator/components/filters/client-auth/src/main/java/org/openrepose/filters/clientauth/openstack/OpenStackAuthenticationHeaderManager.java

```diff
@@ -144,14 +146,22 @@ private void setImpersonator() {
     private void setTenant() {
         filterDirector.requestHeaderManager().putHeader(OpenStackServiceHeader.TENANT_NAME.toString(), cachableToken.getTenantName());
         if(!this.isDelagable && !this.isTenanted) {
-            filterDirector.requestHeaderManager().putHeader(OpenStackServiceHeader.TENANT_ID.toString(), cachableToken.getTenantId());
+            if (sendTenantIdQuality) {
```

Comment:
Then I will take the easy way out and write functional tests. :+1: 

I'm just waiting for the day where we can nuke the Keystone V2 code.